common/src/main/java/net/william278/huskhomes/command/TpCommand.java

```diff
@@ -49,10 +49,21 @@ public void execute(@NotNull CommandUser executor, @NotNull String[] args) {
                             .ifPresent(executor::sendMessage);
                     return;
                 }
-
+                if (!executor.hasPermission(getPermission("other"))) {
+                    plugin.getLocales().getLocale("error_no_permission")
+                            .ifPresent(executor::sendMessage);
+                    return;
+                }
                 this.execute(executor, user, Target.username(args[0]), args);
             }
-            case 2 -> this.execute(executor, Teleportable.username(args[0]), Target.username(args[1]), args);
+            case 2 -> {
+                if (!executor.hasPermission(getPermission("other"))) {
+                    plugin.getLocales().getLocale("error_no_permission")
+                            .ifPresent(executor::sendMessage);
+                    return;
+                }
+                this.execute(executor, Teleportable.username(args[0]), Target.username(args[1]), args);
+            }
```

Comment:
Okay - but what if the player passes their own username as `<username>`? We shouldn't require this permission node for the player to teleport themselves.

We should move these checks to the `execute` logic - checking if 
* the person being teleported is the same as the person executing the teleport
* and, if not, they have this permission node there.